In [ ]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px

df_FGE = pd.read_csv('ForeignGifts_edu.csv', low_memory=False)

df_FGE.head()

,ID,OPEID,Institution Name,City,State,Foreign Gift Received Date,Foreign Gift Amount,Gift Type,Country of Giftor,Giftor Name
0,1,102000,Jacksonville State University,Jacksonville,AL,43738,250000,Monetary Gift,CHINA,NaN
1,2,104700,Troy University,Troy,AL,43592,463657,Contract,CHINA,Confucius Institute Headquarters
2,3,105100,University of Alabama,Tuscaloosa,AL,43466,3649107,Contract,ENGLAND,Springer Nature Customer Service Ce
3,4,105100,University of Alabama,Tuscaloosa,AL,43472,1000,Contract,SAUDI ARABIA,Saudi Arabia Education Mission
4,5,105100,University of Alabama,Tuscaloosa,AL,43479,49476,Contract,SAUDI ARABIA,Saudi Arabia Education Mission


In [3]:
df_FGE.shape

(28221, 10)

In [4]:
df_FGE.columns

Index(['ID', 'OPEID', 'Institution Name', 'City', 'State',
       'Foreign Gift Received Date', 'Foreign Gift Amount', 'Gift Type',
       'Country of Giftor', 'Giftor Name'],
      dtype='object')

In [5]:
df_FGE.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28221 entries, 0 to 28220
Data columns (total 10 columns):
 #   Column                      Non-Null Count  Dtype 
---  ------                      --------------  ----- 
 0   ID                          28221 non-null  int64 
 1   OPEID                       28221 non-null  int64 
 2   Institution Name            28221 non-null  object
 3   City                        28221 non-null  object
 4   State                       28221 non-null  object
 5   Foreign Gift Received Date  28221 non-null  int64 
 6   Foreign Gift Amount         28221 non-null  int64 
 7   Gift Type                   28221 non-null  object
 8   Country of Giftor           28221 non-null  object
 9   Giftor Name                 24470 non-null  object
dtypes: int64(4), object(6)
memory usage: 2.2+ MB


In [7]:
df_FGE['Gift Type'].value_counts()

Gift Type
Contract         17274
Monetary Gift    10936
Real Estate         11
Name: count, dtype: int64

In [8]:
country_total = (
    df_FGE.groupby('Country of Giftor')['Foreign Gift Amount']
    .sum()
    .sort_values(ascending=False)
)

country_total.head(10)

Country of Giftor
QATAR           2706240869
ENGLAND         1464906771
CHINA           1237952112
SAUDI ARABIA    1065205930
BERMUDA          899593972
CANADA           898160656
HONG KONG        887402529
JAPAN            655954776
SWITZERLAND      619899445
INDIA            539556490
Name: Foreign Gift Amount, dtype: int64

### Question 1: Which country gives the most money in total?

**Qatar gives the most money in total, with approximately **$2.71 billion** in foreign gifts.**

In [9]:
country_count = (
    df_FGE.groupby('Country of Giftor')
    .size()
    .sort_values(ascending=False)
)

country_count.head(10)

Country of Giftor
ENGLAND         3655
CHINA           2461
CANADA          2344
JAPAN           1896
SWITZERLAND     1676
SAUDI ARABIA    1610
FRANCE          1437
GERMANY         1394
HONG KONG       1080
SOUTH KOREA      811
dtype: int64

### Question 2: Which country initiates the most gifts by count?

**England initiates the most foreign gifts, with **3,655 gift records** in the dataset.**

In [10]:
country_avg = (
    df_FGE.groupby('Country of Giftor')['Foreign Gift Amount']
    .mean()
    .sort_values(ascending=False)
)

country_avg.head(10)

Country of Giftor
BERMUDA     7.688837e+06
ECUADOR     3.985140e+06
QATAR       3.905109e+06
ANTIGUA     3.435545e+06
JERSEY      3.064589e+06
MOROCCO     2.740214e+06
RUSSIA      2.025462e+06
GUERNSEY    1.750000e+06
IRAQ        1.665520e+06
CYPRUS      1.562512e+06
Name: Foreign Gift Amount, dtype: float64

In [11]:
df_FGE[df_FGE['Country of Giftor'] == 'BERMUDA']['Foreign Gift Amount'].count()

np.int64(117)

In [13]:
df_FGE[df_FGE['Country of Giftor'] == 'BERMUDA']['Foreign Gift Amount'].mean()

np.float64(7688837.367521368)

### Question 3: Which country gives the largest gifts on average?

Bermuda gives the largest gifts on average, with an average foreign gift of approximately **$7.69 million**.

In [15]:
institution_total = (
    df_FGE.groupby('Institution Name')['Foreign Gift Amount']
    .sum()
    .sort_values(ascending=False)
)

institution_total.head(10)

Institution Name
Carnegie Mellon University               1477922504
Cornell University                       1289937761
Harvard University                        954803610
Massachusetts Institute of Technology     859071692
Yale University                           613441311
Texas A&M University                      521455050
Johns Hopkins University                  502409595
Northwestern University                   402316221
Georgetown University                     379950511
University of Chicago (The)               364544338
Name: Foreign Gift Amount, dtype: int64

### Question 4: Which institution receives the most money in total?

Carnegie Mellon University receives the most foreign gift money in total, with approximately **$1.48 billion**.

In [17]:
institution_count = (
    df_FGE.groupby("Institution Name")
    .size()
    .sort_values(ascending=False)
)
institution_count.head(10)

Institution Name
University of California, Los Angeles          3916
California Institute of Technology             2502
Columbia University in the City of New York    1127
Ohio State University (The)                    1014
Johns Hopkins University                        899
University of Michigan - Ann Arbor              715
Stanford University                             668
Cornell University                              653
University of Pennsylvania                      625
Massachusetts Institute of Technology           582
dtype: int64

### Question 5: Which institution receives the greatest number of gifts by count?

The University of California, Los Angeles (UCLA) receives the greatest number of foreign gifts, with **3,916 gift records**. In contrast with Q4: Carnegie Mellon received the most money in total, but UCLA received the most individual gifts.

In [19]:
institution_total = (
    df_FGE.groupby('Institution Name')['Foreign Gift Amount']
    .sum()
)

mean_recieved = institution_total.mean()
median_recieved = institution_total.median()

mean_recieved, median_recieved

(np.float64(52202878.974842764), np.float64(6486791.5))

### Question 6: What is the average amount each university receives? What is the median? Why might they differ?

The mean total amount received by an institution is approximately **$52.2 million**, while the median is approximately **$6.49 million**.

The mean is much larger than the median because the distribution is strongly **right-skewed**. A small number of institutions receive extremely large amounts of foreign funding, which pulls the mean upward. The median is less affected by these very large values and therefore better represents a more typical institution.

In [20]:
university_stats = (
    df_FGE.groupby('Institution Name')['Foreign Gift Amount']
    .agg(['mean', 'median', 'count'])
    .sort_values('mean', ascending=False)
)

university_stats.head(10)

,mean,median,count
Institution Name,,,
Brigham Young University - Idaho,2.270552e+07,22794050.5,4
Western International University,1.294216e+07,417882.0,3
Whittier College,1.207900e+07,937000.0,3
University of Phoenix,1.109435e+07,413241.0,14
Curtis Institute of Music,1.000000e+07,10000000.0,1
Carnegie Mellon University,8.256550e+06,747776.0,179
Williams College,5.222179e+06,1037500.0,14
Brigham Young University,5.217901e+06,617500.0,62
Chamberlain University,5.160818e+06,1018419.0,8


In [21]:
university_stats_filtered = university_stats[
    university_stats['count'] >= 20
]

university_stats_filtered.head(10)

,mean,median,count
Institution Name,,,
Carnegie Mellon University,8.256550e+06,747776.0,179
Brigham Young University,5.217901e+06,617500.0,62
Georgetown University,3.917016e+06,500000.0,97
Northeastern University,2.727380e+06,469238.0,77
Duke University,2.370341e+06,566606.0,145
Boston College,2.333117e+06,750000.0,21
Harvard University,2.107734e+06,893120.0,453
Cornell University,1.975402e+06,250000.0,653
Dartmouth College,1.815049e+06,465273.5,30


In [28]:
country_institution_flow = (
    df_FGE.groupby(
        ['Country of Giftor', 'Institution Name']
    )['Foreign Gift Amount']
    .sum()
    .sort_values(ascending=False)
    .reset_index()
)

country_institution_flow.head(10)

,Country of Giftor,Institution Name,Foreign Gift Amount
0,QATAR,Cornell University,1018473315
1,BERMUDA,Carnegie Mellon University,750000000
2,QATAR,Texas A&M University,504099350
3,QATAR,Carnegie Mellon University,425410702
4,QATAR,Georgetown University,327128310
5,QATAR,Northwestern University,311148564
6,CANADA,Brigham Young University,259927758
7,UNITED ARAB EMIRATES,University of Colorado Boulder,219734835
8,ENGLAND,Harvard University,188328288
9,HONG KONG,Yale University,161131218


In [29]:
country_institution_flow['Foreign Gift Amount'] = (
    country_institution_flow['Foreign Gift Amount']
    .map('${:,.0f}'.format)
)

country_institution_flow.head(10)

,Country of Giftor,Institution Name,Foreign Gift Amount
0,QATAR,Cornell University,"$1,018,473,315"
1,BERMUDA,Carnegie Mellon University,"$750,000,000"
2,QATAR,Texas A&M University,"$504,099,350"
3,QATAR,Carnegie Mellon University,"$425,410,702"
4,QATAR,Georgetown University,"$327,128,310"
5,QATAR,Northwestern University,"$311,148,564"
6,CANADA,Brigham Young University,"$259,927,758"
7,UNITED ARAB EMIRATES,University of Colorado Boulder,"$219,734,835"
8,ENGLAND,Harvard University,"$188,328,288"
9,HONG KONG,Yale University,"$161,131,218"


### Question 7: What is the largest flow from a single country to a single institution?

The largest flow from a single country to a single institution is from **Qatar to Cornell University**, with a total foreign gift amount of approximately **$1.02 billion**.

This result was found by grouping the data by both **Country of Giftor** and **Institution Name**, summing the foreign gift amounts for each country–institution pair, and then sorting the totals from largest to smallest.

In [30]:
top_giftors = (
    df_FGE.groupby('Giftor Name')['Foreign Gift Amount']
    .sum()
    .sort_values(ascending=False)
    .reset_index()
)

top_giftors.head(10)

,Giftor Name,Foreign Gift Amount
0,Qatar Foundation,1166503744
1,Qatar Foundation/Qatar National Res,796197000
2,Qatar Foundation for Education,373945215
3,Anonymous,338793629
4,Saudi Arabian Cultural Mission,275221475
5,HCL,190000000
6,Church of Jesus Christ of LDS,185203715
7,Emirates Institute for Advanced Sc,170641244
8,QIC,148355497
9,Anonymous #9,96334996


### Question 8: Who are the top giftors to U.S. academic institutions?

The largest named giftor in the dataset is the **Qatar Foundation**, with approximately **$1.17 billion** in total foreign gifts.

Other major giftors include **Qatar Foundation/Qatar National Research**, **Qatar Foundation for Education**, the **Saudi Arabian Cultural Mission**, and several anonymous donors.

It is important to note that some giftor names may refer to related organizations under slightly different names, so totals may be split across multiple entries.